spec/lib/dor/update_marc_record_service_spec.rb

```diff
@@ -216,167 +202,128 @@
+      content_metadata_ng_xml = Nokogiri::XML(build_content_metadata_2)
+      identity_metadata_ds = double(Dor::IdentityMetadataDS)
+      content_metadata_ds = double(Dor::ContentMetadataDS)
 
-      allow(content_metadata_xml).to receive_messages(
-        ng_xml: Nokogiri::XML(build_content_metadata_2)
-      )
-
-      d = Dor::Item.new
-
-      allow(d).to receive_messages(
-        datastreams: { 'identityMetadata' => identity_metadata_xml, 'contentMetadata' => content_metadata_xml }
-      )
+      expect(d).to receive(:id).and_return('')
+      expect(d).to receive(:datastreams).twice.and_return({'identityMetadata' => identity_metadata_ds, 'contentMetadata' => content_metadata_ds})
+      expect(identity_metadata_ds).to receive(:ng_xml).and_return(identity_metadata_ng_xml)
+      expect(content_metadata_ds).to receive(:ng_xml).and_return(content_metadata_ng_xml)
 
       updater = Dor::UpdateMarcRecordService.new(d)
+      expect(updater).to receive(:object_type).and_return('|xnotacollection')
       expect(updater.display_type).to eq('|xcitation')
     end
   end
 
   describe '.barcode' do
     it 'should return barcode from a valid identityMetadata' do
-      identity_metadata_xml = double(String)
-
-      allow(identity_metadata_xml).to receive_messages(
-        ng_xml: Nokogiri::XML(build_identity_metadata_1)
-      )
+      d = double(Dor::Item)
+      identity_metadata_ng_xml = Nokogiri::XML(build_identity_metadata_1)
+      identity_metadata_ds = double(Dor::IdentityMetadataDS)
 
-      d = Dor::Item.new
-
-      allow(d).to receive_messages(
-        datastreams: { 'identityMetadata' => identity_metadata_xml }
-      )
+      expect(d).to receive(:id).and_return('')
+      expect(d).to receive(:datastreams).and_return({'identityMetadata' => identity_metadata_ds})
+      expect(identity_metadata_ds).to receive(:ng_xml).and_return(identity_metadata_ng_xml)
 
       updater = Dor::UpdateMarcRecordService.new(d)
       expect(updater.barcode).to eq('|xbarcode:36105216275185')
     end
 
     it 'should return an empty x subfield for identityMetadata without barcode' do
-      identity_metadata_xml = double(String)
-
-      allow(identity_metadata_xml).to receive_messages(
-        ng_xml: Nokogiri::XML(build_identity_metadata_3)
-      )
+      d = double(Dor::Item)
+      identity_metadata_ng_xml = Nokogiri::XML(build_identity_metadata_3)
+      identity_metadata_ds = double(Dor::IdentityMetadataDS)
 
-      d = Dor::Item.new
-
-      allow(d).to receive_messages(
-        datastreams: { 'identityMetadata' => identity_metadata_xml }
-      )
+      expect(d).to receive(:id).and_return('')
+      expect(d).to receive(:datastreams).and_return({'identityMetadata' => identity_metadata_ds})
+      expect(identity_metadata_ds).to receive(:ng_xml).and_return(identity_metadata_ng_xml)
 
       updater = Dor::UpdateMarcRecordService.new(d)
-      expect(updater.object_type).to eq('|x')
+      expect(updater.barcode).to be nil
```

Comment:
i assumed:
* that `barcode` was the actual intended call here, and that `object_type` was a copy-paste holdover or something similar.
* that `barcode`'s current behavior of returning a `nil` (as opposed to `|x`) when there's no barcode is correct.